### PR TITLE
feat(PB): update Cheggaaa's version

### LINKED ISSUE #12433: Update Cheggaaa's PB from v1 to v3
Version 3 of the progress bar library we use (Cheggaaa's "pb") was released  quite a while ago. We should update from `gopkg.in/cheggaaa/pb.v1` to `github.com/cheggaaa/pb/v3`, but will need to handle the breaking API changes.

### PR DESCRIPTION
**Pull request by virajbhartiya**
Closes #12433

Updated the version for Cheggaaa's pb from `v1` to `v3`.

### CODE REVIEW

**Comment by virajbhartiya** [comment]
@rvagg I am unable to understand why [TestF3_Rebootstrap](https://github.com/filecoin-project/lotus/actions/runs/11063981852/job/30741029746?pr=12518#step:9:25233 ) is failing, it's passing when I run it locally.

**Comment by rvagg** [comment]
@virajbhartiya ignore, I've opened an issue for flaky f3 itests: https://github.com/filecoin-project/lotus/issues/12519

**Comment by virajbhartiya** [comment]
I have made the changes required for transitioning from v1 to v3, can you please give it a look @rvagg

**Comment by rvagg** [comment]
@virajbhartiya do you have an easy way to show a screenshot of what it looked like before and after so we know what we're in for? 

Also with these removed:

```go
		bar.ShowTimeLeft = true
		bar.ShowPercent = true
		bar.ShowSpeed = true
		bar.Units = pb.U_BYTES
```

What are the implications? Do we lose functionality?

**Comment by virajbhartiya** [comment]
I based the progress bar format on the `Full ProgressBarTemplate` preset from the `pb` library. This preset is defined in the library's source code:

```go
Full ProgressBarTemplate = `{{with string . "prefix"}}{{.}} {{end}}{{counters . }} {{bar . }} {{percent . }} {{speed . }} {{rtime . "ETA %s"}}{{with string . "suffix"}} {{.}}{{end}}`
```

You can find this definition in the `pb` repository:
https://github.com/cheggaaa/pb/blob/4ca3463f4bcf4446e1746efc6e484e795cfe7b12/v3/preset.go#L5-L6

This preset was suggested as a good starting point in this Slack conversation:
https://filecoinproject.slack.com/archives/CP50PPW2X/p1726604055915309?thread_ts=1726502890.195279&cid=CP50PPW2X

**Comment by virajbhartiya** [comment]
@rvagg 
This was the `v1`
Screenshot 2024-09-27 at 3 57 25 PM

This is the updated version `v3`
Screenshot 2024-09-27 at 3 53 17 PM

File: cmd/lotus-shed/sectors.go [hunk 1]
```diff
@@ -542,20 +542,16 @@ fr32 padding is removed from the output.`,
...
 		l := int64(abi.PaddedPieceSize(length).Unpadded())
 
-		bar := pb.New64(l)
+		bar := pb.Full.Start64(l)
 		br := bar.NewProxyReader(upr)
-		bar.ShowTimeLeft = true
-		bar.ShowPercent = true
-		bar.ShowSpeed = true
-		bar.Units = pb.U_BYTES
-		bar.Output = os.Stderr
```
Comment (Stebalien) [inline cmd/lotus-shed/sectors.go:551]: Stderr is the default now, so removing this is fine.

File: cmd/lotus-shed/sectors.go [hunk 2]
```diff
@@ -542,20 +542,16 @@ fr32 padding is removed from the output.`,
...
-		bar.Start()
 
 		_, err = io.CopyN(os.Stdout, br, l)
 		if err != nil {
+			bar.Finish()
 			return xerrors.Errorf("reading data: %w", err)
 		}
 
+		bar.Finish()
```
Comment (Stebalien) [inline cmd/lotus-shed/sectors.go:554]:
```suggestion
		bar.Finish()
		if err != nil {
			return xerrors.Errorf("reading data: %w", err)
		}

```

**Review by Stebalien — APPROVED** [review]
One change, otherwise LGTM.

File: cmd/lotus-shed/sectors.go (hunk 2 above)
Comment (virajbhartiya) [reply cmd/lotus-shed/sectors.go:554]: Got it, I'll make the change.

**Review by Stebalien — APPROVED** [review]
Thanks!

**Comment by rvagg** [comment]
👌 nice, thanks @virajbhartiya

**Pull request merged**